### PR TITLE
Feature/deprecated filter

### PR DESCRIPTION
**Pull request by cavrob-bodc**
https://jira.ceh.ac.uk/browse/BODCDMSYS-7566
Added a checkbox to filter the search in order to include/exclude deprecated terms, by default deprecated terms are included

### CODE REVIEW

File: src/app/flask_app.py [hunk 1]
```diff
@@ -59,6 +59,12 @@ def process_metadata():
     restrict_to_themes = request.args.get("Restrict to Themes")
     if restrict_to_themes:
         restrict_to_themes = restrict_to_themes.split(",")
+    
+    exclude_deprecated = request.args.get("excludeDeprecated", "false").lower() == "true"
+
+    #exclude_deprecated = True
+    # Print the value of exclude_deprecated to the console
+    print(f"exclude_deprecated: {exclude_deprecated}")
```
Comment (matcor-bodc) [inline src/app/flask_app.py:67]: These lines seem to just be for debugging, I'd remove them

File: src/analyse.py [hunk 2]
```diff
@@ -45,12 +45,12 @@
 }
 
 
-def analyse_from_full_xml(xml_string, restrict_to_themes):        
+def analyse_from_full_xml(xml_string, restrict_to_themes, exclude_deprecated):
```
Comment (matcor-bodc) [inline src/analyse.py:48]: For safety I'd give a default value everywhere a new argument has been added. This and line 90 are the two cases I can see

File: src/app/flask_app.py (hunk 1 above)
Comment (cavrob-bodc) [reply src/app/flask_app.py:67]: Ah yes! forgot to remove them, cheers

File: src/analyse.py (hunk 2 above)
Comment (cavrob-bodc) [reply src/analyse.py:48]: yeah sounds right, will do

**Pull request merged**